src/dtx/dtx_resync.c

```diff
@@ -390,3 +390,58 @@ dtx_resync(daos_handle_t po_hdl, uuid_t po_uuid, uuid_t co_uuid, uint32_t ver,
 	ds_cont_child_put(cont);
 	return rc;
 }
+
+static int
+container_scan_cb(daos_handle_t ih, vos_iter_entry_t *entry,
+		  vos_iter_type_t type, vos_iter_param_t *iter_param,
+		  void *data, unsigned *acts)
+{
+	struct dtx_resync_arg	*arg = data;
+	int			rc;
+
+	rc = dtx_resync(iter_param->ip_hdl, arg->pool_uuid, entry->ie_couuid,
+			arg->version, true);
+	if (rc)
+		D_ERROR(DF_UUID" dtx resync failed: rc %d\n",
+			DP_UUID(arg->pool_uuid), rc);
+
+	return rc;
+}
+
+static int
+dtx_resync_one(void *data)
+{
+	struct dtx_resync_arg	*arg = data;
+	struct ds_pool_child	*child;
+	vos_iter_param_t	param = { 0 };
+	struct vos_iter_anchors	anchor = { 0 };
+	int			rc;
+
+	child = ds_pool_child_lookup(arg->pool_uuid);
+	if (child == NULL)
+		D_GOTO(out, rc = -DER_NONEXIST);
+
+	param.ip_hdl = child->spc_hdl;
+	param.ip_flags = VOS_IT_FOR_REBUILD;
+	rc = vos_iterate(&param, VOS_ITER_COUUID, false, &anchor,
+			 container_scan_cb, NULL, arg);
+
+	ds_pool_child_put(child);
+out:
+	D_DEBUG(DB_TRACE, DF_UUID" iterate pool done: rc %d\n",
+		DP_UUID(arg->pool_uuid), rc);
+
+	return rc;
+}
+
+void
+dtx_resync_ult(void *data)
```

Comment:
Please consider following items before introducing such uncontrolled long-run ULT:

- What if a container is destroyed while iterating the container tree. (re-probe is required);
- What if the container is destroyed while it's iterate callback yielding. (I suppose current resync logic prevent this from happening?)
- How the ULT is informed to abort container resync on container destroy. (I suppose current resync logic did this?)
- How the ULT is informed to abort on pool destroy.